scripts/setup-opam.sh

```diff
@@ -10,65 +16,25 @@ export OPAMYES=1
 # Set term to xterm if not set
 export TERM=${TERM:-xterm}
 
-SWITCH='ocaml-variants.4.07.1+logoom'
-
-if [[ -d ~/.opam ]]; then
-  # ocaml environment
+# init opam
+if ! [[ -d ~/.opam ]]; then
+  opam init
   eval $(opam config env)
-
-  # check for cache'd opam
-  SWITCH_LIST=$(opam switch list -s)
-
-  # Check to see if we have explicit switch version
-  SWITCH_FOUND=false
-  for val in $SWITCH_LIST; do
-    if [ $val == $SWITCH ]; then
-      SWITCH_FOUND=true
-    fi
-  done
-else
-  # if there is no opam switch initialized, start from scratch
-  SWITCH_FOUND=false
 fi
 
-pushd /home/opam/opam-repository && git pull && popd
+# create local switch
+opam switch import src/opam.export --switch .
+sudo chmod -R u+rw _opam
+eval $(opam config env)
 
-if [ "$SWITCH_FOUND" = true ]; then
-  # Add the o1-labs opam repository
-  opam repository add --yes --all --set-default o1-labs https://github.com/o1-labs/opam-repository.git
-  opam switch set $SWITCH
-else
-  # Build opam from scratch
-  opam init
-  # Add the o1-labs opam repository
-  opam repository add --yes --all --set-default o1-labs https://github.com/o1-labs/opam-repository.git
-  opam update
-  opam switch create $SWITCH || true
-  opam switch $SWITCH
-fi
+# add custom O(1) Labs opam repository to local switch
+O1LABS_REPO='https://github.com/o1-labs/opam-repository.git'
+opam repository add --yes --this-switch o1-labs "$O1LABS_REPO"
```

Comment:
made the opam repo local to the switch (instead of global)

Reply:
I think you still want `--set-default`, otherwise `opam` won't look in the repo by default when installing packages.

Reply:
the problem is that `--set-default` sets it as default for all switches IIUC. I think the proposed command is what we want, if I understand correctly it does become the priority for the local switch:

```
$ opam repository list
[NOTE] These are the repositories in use by the current switch. Use '--all' to see all configured repositories.

<><> Repository configuration for switch /Users/davidwong/Work/official_mina
 1 o1-labs git+https://github.com/o1-labs/opam-repository.git
 2 default https://opam.ocaml.org
```

Reply:
This is the correct behavior for local switches. Resolving